### PR TITLE
LensFlarePlugin: Improve copyTexImage2D call

### PR DESCRIPTION
**Pull request by Mugen87**
see #8321

### CODE REVIEW

**Comment by Mugen87** [comment]
I'm not sure if `areaToCopy` is a good expression. Any improvement suggestions?

File: src/renderers/webgl/plugins/LensFlarePlugin.js [hunk 1]
```diff
@@ -337,13 +339,15 @@ THREE.LensFlarePlugin = function ( renderer, flares ) {
 			screenPositionPixels.x = screenPosition.x * halfViewportWidth + halfViewportWidth;
 			screenPositionPixels.y = screenPosition.y * halfViewportHeight + halfViewportHeight;
 
+			areaToCopy.x = viewport.x + screenPositionPixels.x - 8; // horizontal coordinate of the lower left corner of the pixels to copy
+			areaToCopy.y = viewport.y + screenPositionPixels.y - 8; // vertical coordinate of the lower left corner of the pixels to copy
+
 			// screen cull
 
-			if ( hasVertexTexture || (
```
Comment (mrdoob) [inline src/renderers/webgl/plugins/LensFlarePlugin.js]: How come you removed the `hasVertexTexture` check?

Comment (Mugen87) [inline src/renderers/webgl/plugins/LensFlarePlugin.js]: If `hasVertexTexture` is true, the "logical OR" makes the evaluation of the viewport check irrelevant. I think this behavior is not correct. The viewport check should be the only criterion which determines rendering. `hasVertexTexture` is just a flag to load the correct vertex and fragment shader.

Comment (mrdoob) [inline src/renderers/webgl/plugins/LensFlarePlugin.js]: I see I see.

Actually, we may want to remove the `MAX_VERTEX_TEXTURE_IMAGE_UNITS === 0`  code path. WebGLStats says they're only 1.8%.

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
Oh, actually, we could also use `THREE.Box2` for that area check too.

**Comment by Mugen87** [comment]
Yeah, that would be some nice refactoring!

**Pull request merged**